src/main/java/com/mael/ttt/Game.java

```diff
@@ -51,7 +55,7 @@ private void setPlayers() {
     private boolean nextTurn() {
         gameUI.printBoard(board);
 
-        Player currentPlayer = (currentMark.equals("X")) ? player1 : player2;
+        Player currentPlayer = (currentMark.equals(playerMark)) ? player1 : player2;
         int index            = currentPlayer.getCellIndex(board);
         board.setCell(index, currentMark);
 
```

Comment:
There are a few problems in here which make the code not clear to the casual reader:

1.) The ternary operator is usually something you should try avoiding in the code.
2.) Deciding who the `currentPlayer` does not seem like something that `nextTurn` should know how to calculate (`SRP`) same counts for the `index`.

This could possibly be simplified by having a `currentPlayer`method

```
private Player currentPlayer() {
  if currentMark.equals(playerMark)
    return player1;
  else
    return player2;
}
```

When I read this `board.setCell(index, currentMark);` I think you are actually placing the players choice on the board. If so I would actually pass in the player (the current one) and change the methods signature and name:

`board.placeMarkFor(currentPlayer(), currentMark));`